src/main/java/com/github/manolo8/darkbot/gui/login/SavedLogins.java

```diff
@@ -109,14 +111,41 @@ private void updateList() {
 
     private char[] createMasterPassword() {
         if (ConfigEntity.INSTANCE.getConfig().BOT_SETTINGS.OTHER.DISABLE_MASTER_PASSWORD) return new char[]{};
-        JPasswordField pass = new JPasswordField(10);
-
-        JOptionPane.showMessageDialog(this, new Object[]{
-                        "Master password for darkbot to encrypt your credentials.\n" +
-                                "You can use a blank (empty) password:", pass},
-                "Darkbot Master password", JOptionPane.PLAIN_MESSAGE);
-
-        return pass.getPassword();
+        JPanel panel = new JPanel(new MigLayout("ins 0", "[]push[]"));
+        JLabel label = new JLabel("Master password for darkbot to encrypt your credentials.");
+        JPasswordField pass = new JPasswordField(26);
+        JCheckBox check = new JCheckBox("Disable Master Password");
+        JButton button = new JButton("OK");
+        panel.add(label, "span");
+        panel.add(pass, "span");
+        panel.add(check);
+        panel.add(button);
+        button.setEnabled(false);
+
+        check.addItemListener(e -> {
+            if (e.getStateChange() == ItemEvent.SELECTED) {
+                pass.setEditable(false);
+                button.setEnabled(true);
+            } else {
+                pass.setEditable(true);
+                if (new String(pass.getPassword()).isEmpty())
+                    button.setEnabled(false);
+            }
```

Comment:
```suggestion
            pass.setEditable(e.getStateChange() != ItemEvent.SELECTED);
            button.setEnabled(pass.getPassword().length > 0 || e.getStateChange() == ItemEvent.SELECTED);
```

This logic can be cleaned up & simplified a bit